### PR TITLE
Make user-agent for linkchecks customisable

### LINKED ISSUE #2557: Linkcheck failing because of user agent used by Documenter
As discovered in https://github.com/JuliaParallel/MPI.jl/pull/856#issuecomment-2274407303, `MPI.jl` has a linkcheck failing because of the useragent hardcoded in Documenter https://github.com/JuliaDocs/Documenter.jl/blob/c2ed0b7f996bfcafa9f9ff1a5c854ed318fe72b4/src/docchecks.jl#L216-L217
```console
$ curl -I 'https://www.intel.com/content/www/us/en/developer/tools/oneapi/mpi-library.html'
HTTP/2 200
content-type: text/html;charset=utf-8
server: Apache
x-dispatcher: dispatcher2uswest1-b80
x-vhost: publish
last-modified: Wed, 07 Aug 2024 13:09:27 GMT
etag: W/"30f33-61f179eb85c9b-gzip"
expires: Wed, 07 Aug 2024 21:40:56 GMT
cache-control: max-age=0, no-cache
pragma: no-cache
date: Wed, 07 Aug 2024 21:40:56 GMT
set-cookie: detected_bandwidth=LOW; path=/; domain=.intel.com; secure; HttpOnly
set-cookie: src_countrycode=gb; path=/; domain=.intel.com; secure; HttpOnly
alt-svc: h3=":443"; ma=93600
content-security-policy: frame-ancestors 'self' https://premiersupport.intel.com https://c0.avaamo.com *.intel.com; object-src 'self';
x-grn: 0.8d3e1202.1723066856.1047ccba
x-frame-options: SAMEORIGIN
access-control-allow-origin: *
x-xss-protection: 1; mode=block
strict-transport-security: max-age=31536000 ; preload
x-content-type-options: nosniff
$ curl --user-agent "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36" -sI 'https://www.intel.com/content/www/us/en/developer/tools/oneapi/mpi-library.html'
HTTP/2 403
server: AkamaiGHost
mime-version: 1.0
content-type: text/html
content-length: 460
expires: Wed, 07 Aug 2024 21:46:56 GMT
cache-control: max-age=0, no-cache
pragma: no-cache
date: Wed, 07 Aug 2024 21:46:56 GMT
set-cookie: detected_bandwidth=LOW; path=/; domain=.intel.com; secure; HttpOnly
set-cookie: src_countrycode=gb; path=/; domain=.intel.com; secure; HttpOnly
alt-svc: h3=":443"; ma=93600
content-security-policy: frame-ancestors 'self' https://premiersupport.intel.com https://c0.avaamo.com *.intel.com; object-src 'self';
x-grn: 0.8d3e1202.1723067216.1050300f
x-frame-options: SAMEORIGIN
access-control-allow-origin: *
x-xss-protection: 1; mode=block
strict-transport-security: max-age=31536000 ; preload
x-content-type-options: nosniff
```
I'm not sure of what would be a good resolution of this issue, a somewhat customisable user agent?

### PR DESCRIPTION
**Pull request by giordano**
This should fix #2557.  Untested, also because I don't know how to use this new option, but I followed #1295 as a template to follow to add a custom user agent.

### CODE REVIEW

**Comment by mortenpi** [comment]
I think it's a good idea to have an option for this, even though for you use case, #2569 should also fix it. So do I want to merge this too.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,11 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## UNRELEASED
+## UNRELEASD
```
Comment (giordano) [inline CHANGELOG.md:6]:
```suggestion
## UNRELEASED
```

**Review by giordano — COMMENTED** [review]
Thanks!

**Comment by mortenpi** [comment]
Thanks @giordano!

**Pull request merged**